ophyd/signal.py

```diff
@@ -1711,6 +1711,11 @@ def set(self, value, *, timeout=None, settle_time=None):
         --------
         Signal.set
         '''
+        if timeout is DEFAULT_WRITE_TIMEOUT:
+            timeout = self.write_timeout
+            if timeout is None:
```

Comment:
I do not think we want to fallback to the read time out here given that they are independently configurable https://github.com/bluesky/ophyd/blob/33bfc595ac2862aede96f023296f564832144551/ophyd/signal.py#L679-L699

and in the init we have a long comment differentiating the two https://github.com/bluesky/ophyd/blob/33bfc595ac2862aede96f023296f564832144551/ophyd/signal.py#L757-L765

Reply:
We also don't do this fallback in `put` https://github.com/bluesky/ophyd/blob/33bfc595ac2862aede96f023296f564832144551/ophyd/signal.py#L1661-L1665

Reply:
My original version (setting `timeout` from `self.timeout`) was my best understanding of what to do,  Then I learned about `DEFAULT_WRITE_TIMEOUT`.  Now I'm puzzled what to do.  I thought the default timeout that was set in `EpicsSignalBase.set_default_timeout(timeout=10, connection_timeout=5)` would apply in this situation.

Can you suggest the code to use?

Reply:
Perhaps this is the one to use as a fallback? https://github.com/bluesky/ophyd/blob/33bfc595ac2862aede96f023296f564832144551/ophyd/signal.py#L866

Reply:
The reason I'm puzzled is that `self.write_timeout` is `None` in my test case and has not been affected by the `set_default_timeout()` method.  I assumed it would be.  Perhaps fallback (when `write_timeout` is unconfigured) to `__default_timeout ` fits best what I assumed would happen.